sw/device/freestanding/iso646.h

```diff
@@ -0,0 +1,32 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_FREESTANDING_ISO646_H_
+#define SW_DEVICE_FREESTANDING_ISO646_H_
+
+/**
+ * This header implements the iso646.h standard header, as required by C11 S4.6.
+ * See S7.9 of the same for a description.
+ *
+ * This file is provided only for standards compliance reason; including this
+ * file is a style violation.
+ */
+
+#define and &&
+#define and_eq &=
+#define bitand &
+
+#define or ||
+#define or_eq |=
+#define bitor |
+
+#define xor ^
+#define xor_eq ^=
+
+#define not !
+#define not_eq !=
+
+#define compl ~
+
+#endif  // SW_DEVICE_FREESTANDING_ISO646_H_
```

Comment:
ISO646 is gross.  Make it an error to include and `#if 0` the body out so that we know when we're dealing with things that include it?

Reply:
I'd like to hear what other people on this thread have to say about this. I'm not against it, though.

Reply:
> ISO646 is gross. Make it an error to include and `#if 0` the body out so that we know when we're dealing with things that include it?

I second this, but if we will just error out, how about not having this file at all?

Reply:
Well... it's complicated. In practice, this file will never be used, so yeah, deleting it is not wrong in that view. But we seem to have settled on "we are following the ISO standard to the letter except when we absolutely can't." There's certainly an aspect of "we are going to march off this cliff because The Book commands it" to that view, but for a project like OT I think we're better safe than sorry.

This file is small and simple enough that the maintenance cost is actually smaller than the cost of deviating from the ISO standard.